bonsai/BudgetTransactions/BudgetTransactions.swift

```diff
@@ -0,0 +1,181 @@
+                    UITableView.appearance().backgroundColor = .clear
+                }
+            }
+        }
+    }
+}
+
+struct BudgetTransactions_Previews: PreviewProvider {
+    
+    private static let viewContext = DataController.sharedInstance.container.viewContext
+    
+    private static let transactions = [
+        Transaction(context: Self.viewContext,
+                    amount: 20.0,
+                    date: .from(year: 2021, month: 8, day: 28),
+                    category: .init(
+                        context: viewContext,
+                        title: "Restaurant",
+                        color: .green,
+                        icon: BonsaiImage.forkKnifeCircle),
+                    account: .init(
+                        context: viewContext,
+                        title: "BonsaiTest"),
+                    type: .expense,
+                    tags: .init()),
+        
+        Transaction(context: Self.viewContext,
+                    amount: 693.5,
+                    date: .from(year: 2021, month: 8, day: 28),
+                    category: .init(
+                        context: viewContext,
+                        title: "Fuel",
+                        color: .blue,
+                        icon: BonsaiImage.fuelpump),
+                    account: .init(
+                        context: viewContext,
+                        title: "BonsaiTest"),
+                    type: .expense,
+                    tags: .init()),
+        
+        Transaction(context: Self.viewContext,
+                    amount: 776602.0,
+                    date: .from(year: 2021, month: 8, day: 22),
+                    category: .init(
+                        context: viewContext,
+                        title: "Graduation",
+                        color: .red,
+                        icon: BonsaiImage.graduationcapCircle),
+                    account: .init(
+                        context: viewContext,
+                        title: "BonsaiTest"),
+                    type: .expense,
+                    tags: .init()),
+        
+        Transaction(context: Self.viewContext,
+                    amount: 40.0,
+                    date: .from(year: 2021, month: 12, day: 2),
+                    category: .init(
+                        context: viewContext,
+                        title: "Salary",
+                        color: .red,
+                        icon: BonsaiImage.amount),
+                    account: .init(
+                        context: viewContext,
+                        title: "BonsaiTest"),
+                    type: .income,
+                    tags: .init())
+    ]
+    
+    static var previews: some View {
+        BudgetTransactions(
+            transactions: transactions)
+            .previewDevice(PreviewDevice(rawValue: "iPhone 12"))
+            .previewDisplayName("iPhone 12")
+    }
+}
+
+extension Date {
```

Comment:
if it is used for this fake dates only, I suggest to add `fileprivate`
other way it might be better to move this extension to a new file

Reply:
fixed

Reply:
👍